app/controllers/admin/users_controller.rb

```diff
@@ -83,6 +83,27 @@ def activate
     end
   end
 
+  # allow an admin to sign-in as any other user
+
+  def switch_user
+    return unless current_user.admin?
```

Comment:
Maybe `redirect_to(root_path, alert: 'Admin access required to view that page.') and return unless current_user.admin?` like you did in `switch_back`?